partitioner/src/main/java/io/confluent/connect/storage/partitioner/TimeBasedPartitioner.java

```diff
@@ -17,31 +17,47 @@
 package io.confluent.connect.storage.partitioner;
 
 import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Schema.Type;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.connect.errors.ConnectException;
 import org.apache.kafka.connect.sink.SinkRecord;
 import org.joda.time.DateTime;
 import org.joda.time.DateTimeZone;
 import org.joda.time.format.DateTimeFormat;
 import org.joda.time.format.DateTimeFormatter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.lang.reflect.InvocationTargetException;
 import java.util.Locale;
 import java.util.Map;
 
 import io.confluent.connect.storage.common.SchemaGenerator;
 import io.confluent.connect.storage.common.StorageCommonConfig;
+import io.confluent.connect.storage.errors.PartitionException;
 
 public class TimeBasedPartitioner<T> extends DefaultPartitioner<T> {
   // Duration of a partition in milliseconds.
+  private static final Logger log = LoggerFactory.getLogger(TimeBasedPartitioner.class);
   private long partitionDurationMs;
+  private String pathFormat;
   private DateTimeFormatter formatter;
+  protected TimestampExtractor timestampExtractor;
 
   protected void init(long partitionDurationMs, String pathFormat, Locale locale, DateTimeZone timeZone,
                       Map<String, Object> config) {
     delim = (String) config.get(StorageCommonConfig.DIRECTORY_DELIM_CONFIG);
     this.partitionDurationMs = partitionDurationMs;
+    this.pathFormat = pathFormat;
     this.formatter = getDateTimeFormatter(pathFormat, timeZone).withLocale(locale);
     try {
       partitionFields = newSchemaGenerator(config).newPartitionFields(pathFormat);
+      timestampExtractor = newTimestampExtractor(
+          (String) config.get(PartitionerConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG));
+      timestampExtractor.configure(config);
```

Comment:
I think the confusion regarding whether this property can accept a custom class has been addressed in the comment re: the config above.

The user will always give a string. Either one of `Wallclock`, `Record`, or `RecordField` or the canonical class name of a custom class, such as `io.mycomp.mypackage.NestedRecordField`(of course they can still give the canonical name of the predefined class such as `io.confluent.connect.storage.partitioner.WallclockTimestampExtractor` but I don't see any advantage in doing so)

Now whether we treat this property as `Type.STRING` (as we do right now), or `Type.CLASS`, that's something internal. It was easier to treat it as a string for now. If there's a good way with a verifier/recommender to treat it as `Type.CLASS` we can change it later. Let me know if you agree.